qiskit_experiments/library/randomized_benchmarking/rb_experiment.py

```diff
@@ -214,25 +230,68 @@ def _generate_circuit(
-            for meta in circuit.metadata["composite_metadata"]:
-                if meta["physical_qubits"] == self.physical_qubits:
-                    return meta
-        return None
+    @classmethod
+    def _default_transpile_options(cls) -> Options:
+        options = super()._default_experiment_options()
+
+        # Assumes typical basis gates for superconduncting processor
+        options.basis_gates = ["sx", "rz", "x", "cx", "measure"]
+
+        return options
+
+    def _finalize(self):
+        super()._finalize()
+
+        # Set constraints to compute basis gates EPGs from an estimated EPC
+        gate_error_ratio = self.experiment_options.gate_error_ratio
+
+        if gate_error_ratio == "default":
+            num_qubit_default_basis = {
+                1: {"sx": 1.0, "rz": 0.0, "x": 1.0},
+                2: {"cx": 1.0},
+            }
+            qubits = tuple(self.physical_qubits)
+            gate_error_ratio = {
+                (qubits, k): v for k, v in num_qubit_default_basis.get(len(qubits), {}).items()
+            }
+
+        if gate_error_ratio != "skip":
+
+            gates_in_estimates = set(q_gate_tup[1] for q_gate_tup in gate_error_ratio.keys())
+            # Validate if assumed gates are a part to basis gates in the transpile options
+            basis_gates = set(self.transpile_options.basis_gates)
+            if not basis_gates.issuperset(gates_in_estimates):
+                raise ValueError(
+                    f"Assumed gates {gates_in_estimates} is not valid "
+                    f"subset of basis gates {basis_gates}."
+                )
+
+        # Directly copy the value to experiment data metadata via instance state
+        self._gate_error_ratio = gate_error_ratio
 
     def _transpiled_circuits(self) -> List[QuantumCircuit]:
         """Return a list of experiment circuits, transpiled."""
         transpiled = super()._transpiled_circuits()
-        for c in transpiled:
-            meta = self._get_circuit_metadata(c)
-            if meta is not None:
-                c_count_ops = RBUtils.count_ops(c, self.physical_qubits)
-                circuit_length = meta["xval"]
-                count_ops = [(key, (value, circuit_length)) for key, value in c_count_ops.items()]
-                meta.update({"count_ops": count_ops})
+
+        if self._gate_error_ratio == "skip":
+            # Gate errors are not computed, then counting ops is not necessary.
+            return transpiled
+
+        # Compute average basis gate numbers per Clifford operation
+        # This is probably main source of performance regression.
+        # This should be integrated into transpile pass in future.
+        gate_counts_per_clifford = defaultdict(int)
+        total_cliffs = np.sum(self.experiment_options.lengths)
+        for circ in transpiled:
+            for (qubits, instr), count in RBUtils.count_ops(circ, self.physical_qubits).items():
```

Comment:
It would be nice if this counting could be done as a transpiler analysis pass at some point to improve performance

Reply:
Yes, I wrote this in the comment. Probably we can start custom transpiler pass after this PR.